assets/src/edit-story/components/form/switch.js

```diff
@@ -17,159 +17,43 @@
-  position: absolute;
-  appearance: none;
-`;
-
-const Label = styled.label`
-  z-index: 1;
-  width: 50%;
-  text-align: center;
-  opacity: 0.86;
-  cursor: pointer;
-  overflow: hidden;
-  white-space: nowrap;
-  text-overflow: ellipsis;
-  flex: 1;
-  padding: 0px 6px;
-  color: ${({ hasOffset, theme }) =>
-    hasOffset
-      ? theme.colors.bg.primary
-      : theme.DEPRECATED_THEME.colors.bg.divider};
-
-  ${({ disabled }) =>
-    disabled &&
-    `
-    cursor: default;
-    opacity: 0.3;
-  `}
-
-  ${KEYBOARD_USER_SELECTOR} &:focus-within ~ span {
-    background-color: ${({ theme }) =>
-      theme.DEPRECATED_THEME.colors.accent.primary};
-  }
-`;
-
-const SwitchSpan = styled.span`
-  position: absolute;
-  z-index: 0;
-  top: 1px;
-  left: 1px;
-  display: block;
-  width: calc(50% - 3px);
-  height: 28px;
-  border-radius: 100px;
-  background-color: ${({ theme }) => theme.DEPRECATED_THEME.colors.fg.primary};
-  transition: left 0.15s ease-out;
-
-  ${({ hasOffset }) => hasOffset && `left: calc(50% + 2px);`}
-`;
-
-function Switch({ value, disabled, onChange, onLabel, offLabel }) {
-  const handleChange = useCallback(
-    (checked) => {
-      if (onChange) {
-        onChange(checked);
-      }
+import {
+  Switch as BaseSwitch,
+  SwitchPropTypes,
+  useKeyDownEffect,
+} from '../../../design-system';
+
+function Switch({ onChange, value, ...props }) {
+  const switchRef = useRef(null);
+
+  const handleToggleValue = useCallback(
+    (evt) => {
+      onChange(evt, !value);
     },
-    [onChange]
+    [onChange, value]
   );
-  const ref = useRef();
+
   useKeyDownEffect(
-    ref,
-    ['space', 'enter', 'left', 'right'],
-    () => handleChange(!value),
-    [handleChange, value]
+    switchRef,
+    ['space', 'enter', 'left', 'right', 'up', 'down'],
+    handleToggleValue,
+    [handleToggleValue, value]
   );
 
   return (
-    <SwitchContainer ref={ref}>
-      <Label disabled={disabled} hasOffset={Boolean(value)}>
-        {onLabel}
-        <RadioButton
-          disabled={disabled}
-          onChange={() => handleChange(true)}
-          checked={value}
-          value="on"
-        />
-      </Label>
-      <Label disabled={disabled} hasOffset={!value}>
-        {offLabel}
-        <RadioButton
-          disabled={disabled}
-          onChange={() => handleChange(false)}
-          checked={!value}
-          value="off"
-        />
-      </Label>
-      <SwitchSpan hasOffset={!value} />
-    </SwitchContainer>
+    <BaseSwitch ref={switchRef} onChange={onChange} value={value} {...props} />
   );
 }
-
-Switch.propTypes = {
-  onChange: PropTypes.func.isRequired,
-  onLabel: PropTypes.string.isRequired,
-  offLabel: PropTypes.string.isRequired,
-  value: PropTypes.bool,
-  disabled: PropTypes.bool,
-};
-
+Switch.propTypes = SwitchPropTypes;
 Switch.defaultProps = {
-  value: false,
-  disabled: false,
-  onLabel: __('On', 'web-stories'),
   offLabel: __('Off', 'web-stories'),
+  onLabel: __('On', 'web-stories'),
```

Comment:
I prefer inlining these in the component definition above rather than using `defaultProps`. It'll also be [deprecated for function components](https://twitter.com/dan_abramov/status/1133878326358171650) in the future at some point anyway.